collections_api/handler.py

```diff
@@ -20,13 +20,53 @@
 
 def run(event, _context):
     """ Main run routine for module """
-    event['local'] = event.get("local", False)
-    if 'ssm_key_base' not in event and 'SSM_KEY_BASE' in os.environ:
-        event['ssm_key_base'] = os.environ['SSM_KEY_BASE']
-    config = setup_pipeline_config(event)
+    """ Note:  on 10/28/2020 we made a change to step functions that run these lambdas to run step funcitons in parallel.
+        As a result of this change, this module may accept as "event" either a dictionary, or an array of dictionaries. """
+    if isinstance(event, dict):
+        myevent = event
+    else:
+        myevent = _get_appropriate_event_dict(event)
+    myevent['local'] = myevent.get("local", False)
+    if 'ssm_key_base' not in myevent and 'SSM_KEY_BASE' in os.environ:
+        myevent['ssm_key_base'] = os.environ['SSM_KEY_BASE']
+    config = setup_pipeline_config(myevent)
     collections_api_class = CollectionsApi(config)
     collections_api_class.save_collection_details(['aleph', 'archivesspace', 'curate', 'embark'])
-    event['collectionsApiComplete'] = True
+    myevent['collectionsApiComplete'] = True
+    if isinstance(event, dict):
+        event = myevent
+    else:
+        event = _update_original_event(event, myevent)
```

Comment:
Like _get_appropriate_event_dict couldn't you modify _update_original_event(event, myevent) to include this logic? Something like:
```
def _update_original_event(event: list, myevent: dict) -> dict:
    if isinstance(event, dict):
        event = myevent
    else:
        ...
```